src/evmTest/scala/io/iohk/ethereum/vm/utils/EvmTestEnv.scala

```diff
@@ -134,7 +134,7 @@ trait EvmTestEnv {
              sender: Address = defaultSender): ProgramResult[MockWorldState, MockStorage] = {
       val transaction = MockVmInput.transaction(sender, callData, value, gasLimit, gasPrice, Some(contract.address))
       val blockHeader = MockVmInput.blockHeader
-      val pc = ProgramContext[MockWorldState, MockStorage](transaction, blockHeader, world, config)
+      val pc = ProgramContext[MockWorldState, MockStorage](transaction, contract.address, Program(world.getCode(contract.address)), blockHeader, world, config)
```

Comment:
Same here as with the `evm-runner` as the value is not being transfered. As this only affects tests which don't check that the value is transfered this should not stop from merging this PR.